.github/actions/deploy-statefulset-pods/action.yaml

```diff
@@ -0,0 +1,41 @@
+name: deploy-statefulset-pods
+description: delete the deployment pods and create a stateful set pods with hpa
+inputs:
+  default_namespace:
+    default: default
+  working_directory:
+    default: .
+  oc_namespace:
+    default: tnf
+  oc_pod_timeout:
+    description: "Timeout for the `oc wait` command"
+    default: 120s
+  put_deployment_name:
+    description: "The expected value of the test deployment name"
+    default: test
+  put_pod_label_app:
+    description: "The expected value of the test pod's `app` label"
+    default: test
+  put_pod_name_env_var:
+    description: "The environment variable the test pod's name will be saved to"
+    default: PUT_POD_NAME
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Delete the deployment pods
+      run: |
+        make delete-deployment
+      working-directory: ${{ inputs.working_directory }}
```

Comment:
I think this is a bad idea. If we run it after the create-local-test-infra-resources action, it is going to delete part of the infrastructure setup by that action.

If we want to test statefulsets in our CI, I think it would make more sense to run the other step (make create-statefulset) as part of the create-local-test-infra-resources action, without creating a separate one. What do you think?

Reply:
After discussing it, I now understand the idea: we want to run separate tests:

- First, with the CNF deployed by a "Deployment", where the tests are run using the compiled binary
- The, with the CNF deployed by a "StatefulSet", and there we are running tests using the container.

I first thought we wanted to have both deployments and statefulsets running at the same time, which did not match what the action does.

In that case, this is the right approach. Sorry for the noise.